client/src/components/AllClubs.tsx

```diff
@@ -1,10 +1,152 @@
-import NavBar from "./NavigationBar";
+import React, { useState } from 'react';
+import NavBar from './NavigationBar';
+import { Link } from 'react-router-dom';
 
 export default function AllClubs() {
+  const initialClubs = [
+    { name: 'Tennis Club', id: 'tennis' },
+    { name: 'Chess Club', id: 'chess' },
+    { name: 'Reading Club', id: 'reading' },
+    { name: 'Science Club', id: 'science' },
+    { name: 'Art Club', id: 'art' },
+    { name: 'Music Club', id: 'music' },
+    { name: 'Cooking Club', id: 'cooking' },
+    { name: 'Photography Club', id: 'photography' },
+    { name: 'Gaming Club', id: 'gaming' },
+    { name: 'Dance Club', id: 'dance' },
+  ];
+
+  const [clubs, setClubs] = useState(initialClubs);
+  const [joinedClubs, setJoinedClubs] = useState([]);
+  const [query, setQuery] = useState('');
+
+  // Function to handle joining a club
+  const joinClub = (clubId) => {
+    const joinedClub = clubs.find((club) => club.id === clubId);
+    setClubs(clubs.filter((club) => club.id !== clubId));
+    setJoinedClubs([...joinedClubs, joinedClub]);
+  };
+
+  // Function to handle leaving a joined club
+  const leaveClub = (clubId) => {
+    const leftClub = joinedClubs.find((club) => club.id === clubId);
+    setJoinedClubs(joinedClubs.filter((club) => club.id !== clubId));
+    setClubs([...clubs, leftClub]);
+  };
+
+  // Filter clubs based on query
+  const filteredClubs = clubs.filter((club) =>
+    club.name.toLowerCase().includes(query.toLowerCase())
+  );
+
   return (
-    <div>
+    <div style={{ textAlign: 'center' }}>
       <NavBar />
-      All Clubs
+
+      {/* Total Club Count and Search Bar */}
+      <div
+        style={{
+          position: 'absolute',
```

Comment:
Lots of styling in this file that should be moved to a new CSS file like the other

Reply:
Added